### DISCUSSION TITLE
How to avoid self-approval in environment

### DISCUSSION THREAD

**Discussion by hidetatz** [community/community#12241] [Actions]
We want to have a GitHub actions workflow to deploy our application to production, but we want to add an approval step in the workflow.

We've found we can do that by [Environment](https://docs.github.com/en/actions/managing-workflow-runs/reviewing-deployments) feature, but still have a problem.

We want to restrict anyone to run workflow without other person's approval. However, currently if a person who is going to run the workflow is in the approvers, that person can do that without other person's approval by approving the step by themselves.

Is there a way to avoid self-approval in the approval step in workflow?

**Comment by ethomson** [comment]
Hi @hidetatz - thanks for the feedback.  There's not at the moment, you'd need to require two approvals to accomplish this.  The reason that this functionality isn't available at the moment is because many workflow triggers do not have the notion of a person who start it.  For example, workflows triggered on a schedule aren't started by a particular user.  This is good feedback as we continue planning and iterating on environments; thank you!

**Reply by hidetatz** [reply]
@ethomson 
Hi, thank you for your reply!

> you'd need to require two approvals to accomplish this.

I cannot find how this is possible in my repo settings. Can we configure the environment to require two approvals?

<img width="635" alt="Screen Shot 2022-03-02 at 10 49 45" src="https://user-images.githubusercontent.com/60682957/156279284-19302b13-3d8c-498c-bdff-1c3f054a046a.png">

**Reply by hidetatz** [reply]
Sorry, I missed that this is unavailable for now. Thank you for your reply again!

**Reply by adaounrayane** [reply]
hello @ethomson 
It's now possible to have two approvals via the environment feature in github ? 
Regards

**Comment by msandu62** [comment]
Is there any timeline for this feature, to check for self-approval and do not allow it? Or is not even on your new features list?

**Comment by ghost** [comment]
Michiel van Oudheusden has written an article with a workaround to this problem: [Using GitHub Environments approvals without the ability to self approve](https://mindbyte.nl/2021/09/12/using-github-environments-approvals-without-the-ability-to-self-approve.html)

**Comment by domZippilli** [comment]
Requiring multiple approvals would be a nice feature. So would prohibiting self-approvals. Either one would make the approvals more useful to us.

**Comment by dupuy26** [comment]
"There's an app for that" → https://github.com/Raul6469/self-approve

(disclaimer, I haven't tried it, but I think you would want to run this on self-hosted GHA runners)

**Comment by vhiairrassary** [comment]
We would love to have the "avoid self-approval" feature directly available in GitHub. It would ease compliance on our side

**Comment by Hyolog** [comment]
"Prevent self-review" menu has been released.
![image](https://github.com/community/community/assets/14173879/999050e0-30be-420d-9a3b-0d7f32520bba)

Ref : https://github.blog/changelog/2023-10-16-actions-prevent-self-reviews-for-secure-deployments-across-actions-environments/

need to check your repo visibility first (need public)

Unfortunately, GHES doesn't seem to support it yet 😢

**Reply by iamkoby** [reply]
This no longer appears there...
